media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt

```diff
@@ -0,0 +1,328 @@
+     * active. Currently supported actions are as follows:
+     *
+     * * [Play] - start playing a new track
+     * * [Pause] - pause current track
+     * * [Resume] - resumes current track
+     * * [Stop] - stop current track
+     * * [Seek] - change player position to a particular timestamp
+     */
+    sealed class PlayBackAction {
+        /**
+         * Play the track associated with the given [ArticObject].
+         *
+         * Currently, we only support tracks defined in [ArticObject.audioCommentary].
+         *
+         * @see [ExoPlayer.prepare]
+         * @see [Player.setPlayWhenReady]
+         */
+        class Play(val audioFile: ArticObject) : PlayBackAction()
+
+        /**
+         * Pause the current track.
+         *
+         * You can continue where you left off with [Resume].
+         *
+         * @see [Player.setPlayWhenReady]
+         */
+        class Pause : PlayBackAction()
+
+        /**
+         * Resume the current track.
+         *
+         * You can also resume (without resetting the stream) by sending a subsequent [Play]
+         * action backed by the same [ArticObject].
+         *
+         * @see [Player.setPlayWhenReady]
+         */
+        class Resume : PlayBackAction()
+
+        /**
+         * Stop playback, seek back to the start of the file, and move this service into the background.
+         *
+         * @see [Player.stop]
+         */
+        class Stop : PlayBackAction()
+
+        /**
+         * Move playback to a specific temporal position.
+         *
+         * @param time number of milliseconds from the start of the track
+         */
+        class Seek(val time: Long) : PlayBackAction()
+    }
+
+    /**
+     * Current 'state' of [the ExoPlayer][player].
+     *
+     * We enforce the following states:
+     * * [Playing]
+     * * [Paused]
+     * * [Stopped]
+     */
+    sealed class PlayBackState(val articAudioFile: ArticAudioFile) {
+        class Playing(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+        class Paused(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+        class Stopped(articAudioFile: ArticAudioFile) : PlayBackState(articAudioFile)
+    }
+
+    private val binder: Binder = AudioPlayerServiceBinder()
+    private lateinit var playerNotificationManager: PlayerNotificationManager
+    var articObject: ArticObject? = null
+
+    private val audioControl: Subject<PlayBackAction> = BehaviorSubject.create()
+    val audioPlayBackStatus: Subject<PlayBackState> = BehaviorSubject.create()
+    val currentTrack: Subject<ArticAudioFile> = BehaviorSubject.create()
+
+    val disposeBag = DisposeBag()
+
+    /**
+     * Returns an intent including the details of currently playing audio file.
+     */
+    fun getIntent(): Intent {
+        val audioIntent = "edu.artic.audio".asDeepLinkIntent()
+        audioIntent.putExtra("artic_object", articObject)
```

Comment:
Make this a reusable constant, but it looks like its not being used anywhere?

Reply:
The method does not exist now.